includes/REST_API/Stories_Page_Template_Controller.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/**
+ * Class Stories_Page_Template_Controller
+ *
+ * @package   Google\Web_Stories
+ * @copyright 2021 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://github.com/google/web-stories-wp
+ */
+
+/**
+ * Copyright 2021 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Web_Stories\REST_API;
+
+use stdClass;
+use WP_Error;
+use WP_Post;
+use WP_REST_Posts_Controller;
+use WP_REST_Request;
+use WP_REST_Response;
+/**
+ * Class Stories_Page_Template_Controller
+ *
+ * @package Google\Web_Stories\REST_API
+ */
+class Stories_Page_Template_Controller extends WP_REST_Posts_Controller {
```

Comment:
This is a good opportunity to take a closer look at our REST controllers.

We have:

1. `Stories_Base_Controller` - base class for other post type REST controllers. Mainly adds `story_data` field.  
  Does way too much at the moment. For example `::prepare_links()` and `featured_media_url` should be moved to `Stories_Controller` as it's only relevant for the `web-story` post type, not all post types.
2. `Templates_Controller` (RIP) - we had this one back in the day for the `web-story-template` post type, but it didn't do anything beyond what `Stories_Base_Controller` offered
3. `Stories_Controller` - extends base and also exposes web-story specific things like publisher logo, style presets, etc.

We should change `Stories_Base_Controller` so that it can also serve as a base for this controller here.

Reply:
> We should change Stories_Base_Controller so that it can also serve as a base for this controller here.

This is what a recommended originally, but in our last meeting you recommended a new controller. 
"It needs it's own controller". 

I will look at refactoring.